internal/controllers/openshift.go

```diff
@@ -0,0 +1,199 @@
+// is contributed by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create
+// derivative works of, display, perform, and distribute the Software and make,
+// use, sell, offer for sale, import, export, have made, and have sold the
+// Software and the Larger Work(s), and to sublicense the foregoing rights on
+// either these or other terms.
+//
+// This license is subject to the following condition:
+// The above copyright notice and either this complete permission notice or at
+// a minimum a reference to the UPL must be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	operatorv1beta1 "github.com/cryostatio/cryostat-operator/api/v1beta1"
+	"github.com/cryostatio/cryostat-operator/internal/controllers/common"
+	configv1 "github.com/openshift/api/config/v1"
+	consolev1 "github.com/openshift/api/console/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func (r *CryostatReconciler) reconcileOpenShift(ctx context.Context, cr *operatorv1beta1.Cryostat) error {
+	if !r.IsOpenShift {
+		return nil
+	}
+	err := r.reconcileConsoleLink(ctx, cr)
+	if err != nil {
+		return err
+	}
+	return r.addCorsAllowedOriginIfNotPresent(ctx, cr)
+}
+
+func (r *CryostatReconciler) finalizeOpenShift(ctx context.Context, cr *operatorv1beta1.Cryostat) error {
+	if !r.IsOpenShift {
+		return nil
+	}
+	err := r.deleteConsoleLink(ctx, newConsoleLink(cr))
+	if err != nil {
+		return err
+	}
+	return r.deleteCorsAllowedOrigins(ctx, cr)
+}
+
+func newConsoleLink(cr *operatorv1beta1.Cryostat) *consolev1.ConsoleLink {
+	// Cluster scoped, so use a unique name to avoid conflicts
+	return &consolev1.ConsoleLink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: common.ClusterUniqueName(cr),
+		},
+	}
+}
+
+func (r *CryostatReconciler) reconcileConsoleLink(ctx context.Context, cr *operatorv1beta1.Cryostat) error {
+	link := newConsoleLink(cr)
+
+	url := cr.Status.ApplicationURL
+	if len(url) == 0 {
+		// Nothing to link to, so remove the link if it exists
+		err := r.deleteConsoleLink(ctx, link)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+	op, err := controllerutil.CreateOrUpdate(ctx, r.Client, link, func() error {
+		link.Spec.Link = consolev1.Link{
+			Text: "Cryostat",
```

Comment:
This will patched downstream to something like `Red Hat build of Cryostat`?

Reply:
Good point. I've modified our generated code to include a constant for the application name that's configurable at build time.